core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -509,11 +474,14 @@ protected ReadOnlyKeyCommand remoteReadOnlyCommand(InvocationContext ctx, ReadOn
       return new TxReadOnlyKeyCommand(command, getMutationsOnKey((TxInvocationContext) ctx, command.getKey()));
    }
 
+   protected RpcOptions createRpcOptions() {
+      return rpcManager.getRpcOptionsBuilder(ResponseMode.SYNCHRONOUS_IGNORE_LEAVERS, DeliverOrder.NONE).build();
```

Comment:
doing a search, only the ```BaseRpcInterceptor``` is caching and it shouldn't (```staggeredOptions``` and ```defaultSyncOptions```). 
Async rpc options doesn't matter since the timeout is ignored.